File: 06. Random Forest (Regresion).py
# -*- coding: utf-8 -*-
"""
Created on Thu Aug  5 16:06:32 2021

@author: Admin
"""

## Random Forest

# Usado para regresión

# =============================================================================
# Libraries
# =============================================================================

import pandas as pd
#import numpy as np
import matplotlib.pyplot as plt
import sklearn.ensemble as ens

# =============================================================================
# Read data
# =============================================================================

position_salaries = pd.read_csv("data\Position_Salaries.csv")

x = position_salaries.iloc[:, 1:2].values # es mejor que siempre sea matriz
y = position_salaries.iloc[:, 2].values

# =============================================================================
# Ajustar Random Forest
# =============================================================================

regresion = ens.RandomForestRegressor(n_estimators = 100,
                                      random_state = 0)
regresion.fit(x, y)

# =============================================================================
# Visualizacion
# =============================================================================

plt.scatter(x, y, color = "darkred")
plt.plot(x, regresion.predict(x), color = "darkblue")
plt.title("Modelo CART (Regresion)")
plt.xlabel("Posición del empleado")
plt.ylabel("Sueldo en $")
plt.show()

# =============================================================================
# Prediccion
# =============================================================================

regresion.predict([[6.5]])

File: 06. Random Forest (Regresion).R
## Random Forest ## 

# Para regresión


# libraries ---------------------------------------------------------------

library(tidyverse)
library(randomForest)

# read data ---------------------------------------------------------------

position_salaries <- readr::read_csv("data/Position_Salaries.csv") %>% 
  janitor::clean_names()

position_salaries <- position_salaries %>% 
  dplyr::select(-position)

# ajuste del modelo -------------------------------------------------------

set.seed(1234)

regresion <- position_salaries %>% 
  randomForest::randomForest(salary ~ ., data = .)

# visualizacion -----------------------------------------------------------

position_salaries %>% 
  dplyr::mutate(ajustados = predict(regresion)) %>% 
  ggplot2::ggplot(aes(level, y = salary)) + 
  geom_point(color = "darkred") +
  geom_line(aes(y = ajustados), color = "darkblue") +
  labs(title = "Random Forest (Regresion)",
       x = "Nivel del empleado",
       y = "Salario ($)") +
  theme_classic()


# predicción --------------------------------------------------------------

predict(regresion, data.frame(level = 6.5))

